src/components/Model3DViewer/Model3DViewer.tsx

```diff
@@ -235,21 +284,238 @@ export class Model3DViewer extends React.Component<Model3DViewerProps> {
     }
   };
 
+  slice = (sliceProps: SlicingProps) => {
+    if (this.viewer) {
+      const planes = this.state.planes;
+      if (sliceProps.x) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(sliceProps.x.direction ? 1 : -1, 0, 0),
+          sliceProps.x.coord
+        );
+        planes[0] = plane;
+      }
+      if (sliceProps.y) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, sliceProps.y.direction ? 1 : -1, 0),
+          sliceProps.y.coord
+        );
+        planes[1] = plane;
+      }
+      if (sliceProps.z) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, 0, sliceProps.z.direction ? 1 : -1),
+          sliceProps.z.coord
+        );
+        planes[2] = plane;
+      }
+      this.setState({ planes });
+    }
+  };
+
+  takeScreenShot = async () => {
+    if (this.viewer) {
+      const url = await this.viewer.getScreenshot();
+      if (this.props.onScreenshot) {
+        this.props.onScreenshot(url);
+      }
+    }
+  };
+
+  onchange = (val: number, axis: string) => {
```

Comment:
Rename to `onChange`.
Could we use ENUMs instead of string for `axis`? So the user don't mistype using `X` instead of `x` etc.
Or use `axis: 'x' | 'y' | 'z' or similar

Reply:
Renamed to onChange.
This method is not actually for the user to use, so I changed the type of axis to number so that the switch statement below can be removed without having any axisToIndexMap.